src/package/packageUninstall.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import * as os from 'os';
+import { Connection, Lifecycle, Messages, PollingClient, SfError, StatusResult } from '@salesforce/core';
+import { Duration } from '@salesforce/kit';
+import { PackagingSObjects } from '../interfaces';
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/packaging', 'messages');
+
+type UninstallResult = PackagingSObjects.SubscriberPackageVersionUninstallRequest;
+
+async function poll(id: string, conn: Connection): Promise<StatusResult> {
+  const uninstallRequest = await conn.tooling.sobject('SubscriberPackageVersionUninstallRequest').retrieve(id);
+
+  switch (uninstallRequest.Status) {
+    case 'Success': {
```

Comment:
Is this the actual casing returned by the server?  I'm guessing it is but that's not consistent with the install API, which returns status in all caps.  🙄

Reply:
it is :(